arm/Microsoft.ContainerService/managedClusters/deploy.bicep

```
@@ -128,6 +128,12 @@ param agentPools array = []
 @description('Optional. Specifies whether the httpApplicationRouting add-on is enabled or not.')
 param httpApplicationRoutingEnabled bool = false
 
+@description('Optional. Specifies whether the ingressApplicationGateway (AGIC) add-on is enabled or not.')
+param ingressApplicationGatewayEnabled bool = false
+
+@description('Conditional. Specifies the resource ID of conected application gateway. Must be set if `ingressApplicationGatewayEnabled` is set to `true`.')
```

For consistency with other Conditional descriptions I'd suggest 
```suggestion
@description('Conditional. Required if `ingressApplicationGatewayEnabled` is set to `true`. Specifies the resource ID of connected application gateway.')
```